### PR TITLE
feature: update CoWTrade

### PR DESCRIPTION
**Pull request by adamazad**
# Summary

Updates and `CoWTrade` to the newest Cow SDK. Addresses #1255 by sending back the `quoteId` to `POST /order` endpoint. This PR uses the RC v1.6.0 of the SDK.

### CODE REVIEW

**Review by jorgedclopes — APPROVED** [review]
LGTM

**Comment by MilanVojnovic95** [comment]
If I try to swap from XDAI to USDC or USDT I don't get COW swap as an option in the exchange list
and when I try from USDC to 1INCH I get this error message.

https://user-images.githubusercontent.com/57331244/181448688-5d41385d-bc8f-472f-be13-52d438915423.mp4

**Comment by adamazad** [comment]
@MilanVojnovic95 can you give it another test? Thanks

**Comment by niemam29** [comment]
After i've approved some token and tried to swap the error message says that i've got insufficient funds (i used the max button soo it shouldnt happen), also after changing some tokens swapr crashed. This may be hard do recreate, soo if you face any problems please let me know. 

[screen-capture.webm](https://user-images.githubusercontent.com/30408272/181487030-9e6451e8-db44-41a2-beed-098f5a974f6b.webm)

Logs: 
[logs.txt](https://github.com/levelkdev/swapr-dapp/files/9208394/logs.txt)

**Comment by Mi-Lan** [comment]
Screenshot 2022-08-02 at 20 27 02
Screenshot 2022-08-02 at 20 27 37
Seems that there is some issue when trying to swap cow...

**Comment by Mi-Lan** [comment]
Seems that its for every pair..

https://user-images.githubusercontent.com/33226956/182448515-bf1df8eb-1d02-4c40-a4e3-efba19f22a93.mov

.

File: src/state/transactions/updater.tsx
Comment (mix1o) [inline src/state/transactions/updater.tsx:85]: `res` has type `any`. If possible you can type it

File: src/state/transactions/updater.tsx
Comment (mix1o) [inline src/state/transactions/updater.tsx:86]:
```suggestion
        if (!res) {
```

File: src/state/transactions/updater.tsx
Comment (mix1o) [inline src/state/transactions/updater.tsx:114]: we don't need it

File: src/state/transactions/updater.tsx
Comment (mix1o) [inline src/state/transactions/updater.tsx:123]: should avoid `any`

**Review by mix1o — COMMENTED** [review]
Looks nice!

File: src/state/transactions/updater.tsx
Comment (adamazad) [reply src/state/transactions/updater.tsx:123]: `settlementContract` is no longer in `orderMetadata`. Good catch!

File: src/state/transactions/updater.tsx
Comment (adamazad) [reply src/state/transactions/updater.tsx:85]: Could be due to IntelliSense not picking up the types. The types come from `@cowprotocol/cow-sdk`

image

File: src/state/transactions/updater.tsx
Comment (adamazad) [reply src/state/transactions/updater.tsx]: Addressed in [6ed22de](https://github.com/levelkdev/swapr-dapp/pull/1280/commits/6ed22de5368f9372fd2fc6d703d13e4a65f669df)

File: src/state/transactions/updater.tsx
Comment (adamazad) [reply src/state/transactions/updater.tsx]: [6ed22de](https://github.com/levelkdev/swapr-dapp/pull/1280/commits/6ed22de5368f9372fd2fc6d703d13e4a65f669df)

File: src/state/transactions/updater.tsx
Comment (adamazad) [reply src/state/transactions/updater.tsx]: [6ed22de](https://github.com/levelkdev/swapr-dapp/pull/1280/commits/6ed22de5368f9372fd2fc6d703d13e4a65f669df)

**Review by Diogomartf — APPROVED** [review]

**Comment by Diogomartf** [comment]
Tested swaping xDai to aave and worked great. ✔

**Comment by adamazad** [comment]
Great catch @Mi-Lan. The issue was fixed in https://github.com/levelkdev/swapr-sdk/pull/180.

I'm waiting for #1287 to be merged to avoid confusion about the scope. Then, release the next version of the SDK

**Pull request merged**
